core/src/main/scala/kafka/log/Log.scala

```diff
@@ -180,6 +180,17 @@ object RollParams {
   }
 }
 
+sealed trait LogStartIncrementCause
```

Comment:
How about LogStart```Offset```IncrementCause ?

Reply:
I knew someone was going to ask this 🙂 . I guess I'll do it.